config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -383,7 +384,11 @@ public static Config defaultReference(ClassLoader loader) {
      * @return the default override configuration
      */
     public static Config defaultOverrides() {
-        return systemProperties();
+        if (!getOverrideWithEnv()) {
```

Comment:
Think it's easier to read if you reverse the if/else to avoid the `!`

Reply:
done